e2e/fixtures/fdb_cluster.go

```diff
@@ -61,6 +61,12 @@ func (fdbCluster *FdbCluster) GetFDBImage() string {
 
 // GetSidecarImageForVersion return the sidecar image used for the specified version.
 func (fdbCluster *FdbCluster) GetSidecarImageForVersion(version string) string {
+	// In the case of the unified image the sidecar will also be the main container image.
+	if fdbCluster.cluster.UseUnifiedImage() {
+		return fdbv1beta2.SelectImageConfig(fdbCluster.GetClusterSpec().MainContainer.ImageConfigs, version).
```

Comment:
if not a huge pain, it might be clearer to just set the sidecar config in `fdbCluster.GetClusterSpec()` to what it should be for the unified image
i.e. I think it would make sense in GetSidecarContainerOverrides take a useUnifiedImage param and just set to the main container so other checks like this don't have to do this special logic

Reply:
The operator will use the `MainContainer` config when the unified image is used, so I think it's better to use the `MainContainer` config here to reflect the expected configuration.